### PR TITLE
fix(core): allow canceling `data-tauri-drag-region` maximization on macOS, closes #8306

### LINKED ISSUE #8306: [bug] [macOS] Tauri drag handler should wait for mouse up after double-click-to-maximize
### Describe the bug

After a drag area (titlebar) is double clicked on macOS, Tauri does not wait for the user to release their mouse before maximizing.

https://github.com/tauri-apps/tauri/assets/75394401/3c2104ef-3f02-4e59-880d-ae43ba6a6ad3

### Reproduction

With Tailwind:
```html
<div class="absolute top-0 left-0 right-0 h-10 bg-red-500" data-tauri-drag-region></div>
```

### Expected behavior

After double clicking, maximization is delayed until the mouse button is released. If the mouse button is moved before being released, the maximization is aborted.

https://github.com/tauri-apps/tauri/assets/75394401/5b8c836e-8e69-43bc-a2f0-59cfc80f5ca0

### Platform and versions

```text
[✔] Environment
    - OS: Mac OS 14.0.0 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.74.0 (79e9716c9 2023-11-13)
    ✔ cargo: 1.74.0 (ecb9851af 2023-10-18)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-aarch64-apple-darwin (environment override by RUSTUP_TOOLCHAIN)
    - node: 20.8.0
    - pnpm: 8.10.5
    - yarn: 1.22.21
    - npm: 10.1.0
    - bun: 1.0.6

[-] Packages
    - tauri [RUST]: 1.5.2
    - tauri-build [RUST]: 1.5.0
    - wry [RUST]: 0.24.6
    - tao [RUST]: 0.16.5
    - tauri-cli [RUST]: 1.5.6
    - @tauri-apps/api [NPM]: 1.5.1
    - @tauri-apps/cli [NPM]: 1.5.6

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: Svelte
    - bundler: Vite
```

### Stack trace

_No response_

### Additional context

The current behavior is expected on Windows. I haven't tested on any Linux distros.

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: .changes/tauri-data-drag-region-macos-maximize.md
Comment (amrbashir) [inline .changes/tauri-data-drag-region-macos-maximize.md]:
```suggestion
'tauri': 'patch:bug'
```

**Comment by pewsheen** [comment]
From https://github.com/tauri-apps/tauri/issues/8306#issuecomment-1828900300, we need to save the mouse position on the first click and compare it to the second one

**Comment by amrbashir** [comment]
gonna wait for @lucasfernog to approve this one as I am not sure if this is considered breaking change or not

File: core/tauri/scripts/core.js
Comment (lucasfernog) [inline core/tauri/scripts/core.js]:
```suggestion
  // moves after the double click, it should be cancelled (see https://github.com/tauri-apps/tauri/issues/8306)
```

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**
